geonode/people/views.py

```diff
@@ -189,6 +190,30 @@ def get_queryset(self):
         queryset = self.get_serializer_class().setup_eager_loading(queryset)
         return queryset.order_by("username")
 
+    def perform_create(self, serializer):
+        user = self.request.user
+        if not (user.is_superuser or user.is_staff):
+            raise PermissionDenied()
+        serializer.is_valid(raise_exception=True)
+        instance = serializer.save()
+        return instance
+
+    def update(self, request, *args, **kwargs):
+        instance = self.get_object()
+        serializer = self.get_serializer(instance, data=request.data, partial=True)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data)
+
+    def destroy(self, request, *args, **kwargs):
+        instance = self.get_object()
+        self.perform_destroy(instance)
+        return Response("User deleted sucessfully", status=200)
+
+    def perform_destroy(self, instance):
+        call_user_deletion_rules(instance)
```

Comment:
@RegisSinjari this call should be inside a try/catch and it's its responsibility to return a `PermissionDenied` in case.
I would also rename the method to `check_user_deletion_rules`